tests/platform_tests/test_auto_negotiation.py

```diff
@@ -0,0 +1,358 @@
+import logging
```

Comment:
Suggest to add a header to this file with a summary of what is the motivation and what will be tested.
If you have an HLD please refer to it as well. if not, just short description